models/users.lua

```diff
@@ -1,8 +1,53 @@
+local bcrypt = require ("bcrypt")
+local config   = require("lapis.config").get()
+local token = config.secret
+local trim = require("lapis.util").trim_filter
 local Model = require("lapis.db.model").Model
 local Users = Model:extend("users", {
   relations = {
     {"packages", has_many = "Packages"}
   }
 })
 
+function Users:create_user(user)
+  trim(user, {"name", "password", "admin", "email"}, nil)
+  local hash = bcrypt.digest(user.name .. user.password .. token, config.salt)
+  user.password = nil
+  local user = self:create {
+    name = user.name,
+    password = hash,
+    admin = user.admin,
+    email = user.email
+  }
+  if user then
+    return user
+  end
+  return false, {"err_create_user", {user.name}}
+end
+
+function Users:verify_user(params)
+  local user = self:get_user(params.name)
+  if not user then
+    return false, {"err_invalid_user"}
```

Comment:
Same here and in other spots

Reply:
the dangers of leaning on examples. This may be correct, but I will double check it to make sure I haven't done too much copypasta

Reply:
Figured it out, copied the pattern from my example a little more closely than necessary, the push later tonight will clean this all up